### PR TITLE
Add drag-and-drop.html

### PR DESCRIPTION
**Pull request by shoaibm-aurea**
https://jira.devfactory.com/browse/E2EPLTFRM-5521

### CODE REVIEW

**Review by ajaysuwalka — APPROVED** [review]

**Comment by alexander-a-chernyshev** [comment]
Your Pull Request has been selected for a project-agnostic code review by [Aurea Central Reviewers Team](https://github.com/orgs/trilogy-group/teams/aurea-central-reviewers/members).

Please consider using [Analyzer UI](https://confluence.devfactory.com/display/CodeReview/Analyzer+UI) extension to catch issues in pull requests.

<details><summary>More information</summary>

_This is an initiative to improve overall code quality. If your change is **urgent**, you **don't** have to wait for our review._ 

_NOTE: This review **does not** replace your team's review._

</details>

File: drag-and-drop.html [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+                            revert: true,
+                            revertDuration: 0
+                        });
+                        el.data('event', {title: event.title, id: event.id, stick: true});
+                    }
+                }
+            });
+
+            var isEventOverDiv = function (x, y) {
+                var external_events = $('#external-events');
```
Comment (alexander-a-chernyshev) [inline drag-and-drop.html:122]: <details>
<summary>:warning: <strong>This code contains an identifier named using non-standard conventions.</strong> (click for details)</summary>

A possible improvement would be to follow standard JavaScript naming conventions.  
  
This change would ensure the code follows well-known standards, making it easier for new developers to start without the need to learn a different convention and, thus, improving maintainability.

**[More details](https://confluence.devfactory.com/x/sHTlFQ)**

## The improvement suggestion in this case
```html  
var externalEvents = $('#external-events');
```  
> NOTE: Consider applying the structure of this solution in other parts of the code, if there is another occurrence(s).
</details>

[//]: # (<a id="0b136e61-4250-43c8-a32d-0bcb626708df" data-finding-id="367359152" href="none" codereview-uuid-reserved-tag data-business-id="32d5ffc5-52e8-422b-9061-1bd1da1f2858" data-analyzer="js" />)

File: drag-and-drop.html [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+                    if (isEventOverDiv(jsEvent.clientX, jsEvent.clientY)) {
+                        $('#calendar').fullCalendar('removeEvents', event._id);
+                        var el = $("<div class='fc-event'>").appendTo('#external-events-listing').text(event.title);
+                        el.draggable({
+                            zIndex: 999,
+                            revert: true,
+                            revertDuration: 0
+                        });
+                        el.data('event', {title: event.title, id: event.id, stick: true});
```
Comment (alexander-a-chernyshev) [inline drag-and-drop.html:116]: <details>
<summary>:warning: <strong>Formatting related to spacing is inconsistent.</strong> (click for details)</summary>

A possible improvement would be to apply the current language Coding Conventions (or project specific style) to all files and fix any possible spacing issue before committing to maintain code style consistency.  
  
This change would ensure that the code is consistently clean, thus, improving readability, making variable and block scopes easier to understand.

**[More details](https://confluence.devfactory.com/x/bHvlFQ)**

## The improvement suggestion in this case
```html  
el.data('event', { title: event.title, id: event.id, stick: true });
```  
> NOTE: Consider applying the structure of this solution in other parts of the code, if there is another occurrence(s).
</details>

[//]: # (<a id="e24c511c-6f4d-4b64-98d1-39aaddc71ad0" data-finding-id="367360876" href="none" codereview-uuid-reserved-tag data-business-id="32d5ffc5-52e8-422b-9061-1bd1da1f2858" data-analyzer="js" />)

File: drag-and-drop.html [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+                offset.bottom = external_events.height() + offset.top;
+
+                // Compare
+                if (x >= offset.left
+                    && y >= offset.top
+                    && x <= offset.right
+                    && y <= offset.bottom) {
+                    return true;
+                }
+                return false;
```
Comment (alexander-a-chernyshev) [inline drag-and-drop.html:134]: <details>
<summary>:warning: <strong>This code is missing a blank line after the code block and before the next statement or before the comment.</strong> (click for details)</summary>

Consider leaving a single blank line after blocks/group of statements and before the next statement or before the comment unless it's on the first line of a block.  
  
Adding a blank line improves code readability and keeps the style consistent.

**[More details](https://confluence.devfactory.com/x/hnvlFQ)**

## The improvement suggestion in this case
Consider adding a single blank line before the statement or comment.  
> NOTE: Consider applying the structure of this solution in other parts of the code, if there is another occurrence(s).
</details>

[//]: # (<a id="e278650c-65a3-4282-abc9-95317b2a3ff6" data-finding-id="367360902" href="none" codereview-uuid-reserved-tag data-business-id="32d5ffc5-52e8-422b-9061-1bd1da1f2858" data-analyzer="js" />)

**Review by alexander-a-chernyshev — COMMENTED** [review]
Hi, I am the JavaScript expert who performed the review of this Pull Request. We have well-defined [coding and review standards](https://confluence.devfactory.com/display/CodeReview/Findings), and you will find below suggestions to improve the code quality, based on these guidelines.
My additional comments are in the Notes section. I hope you find them useful.

<table>
  <tr>
    <td>:no_entry_sign:<b>Severes:</b>  0</td>
    <td>:exclamation:<b>Reds:</b>  0</td>
    <td>:warning:<b>Yellows:</b>  3</td>
    <td>:white_check_mark:<b>Greens:</b>  0</td>
  </tr>
</table>

<details><summary>More information</summary>
  <table>
    <tr>
      <td>:no_entry_sign:</td>
      <td><b>Severe Findings</b> point out serious violations which can affect stability, performance, maintainability and therefore should be fixed.
        <ul>
          <li>Please note that the merge might have been blocked until all Severe Findings are fixed.</li>
          <li>While it is highly recommended to implement all findings only Severe Findings might have been blocking the merge</li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>:exclamation:</td>
      <td><b>Red Findings</b> represent potential bugs and it is highly recommended to implement them.<br/>
      Not required for the approval of the PR
      </td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td><b>Yellows</b> are suggestions to improve the code, warnings and tips<br/>
      Not required for the approval of the PR, and should not be used for FTAR calculation.
      </td>
    </tr>
    <tr>
      <td>:white_check_mark:</td>
      <td><b>Green Findings</b> highlight good ideas and solutions</td>
    </tr>
    <tr>
      <td>:repeat:</td>
      <td><b>Repeated</b> instances of the same findings are counted just once and flagged with the repeat icon</td>
    </tr>
  </table>

  Feel free to reply to this comment, (make sure you `@tag` us) so we can start a conversation. The goal here is to learn and grow together and make our products better.

  > NOTE: This review **does not** replace your team's review._
</details>
<details>
    <summary>T4W FTAR Statistics</summary>
    <br />
    CodeRead team introduced new quality measurement to evaluate the trailing 4 weeks First Time Acceptance Rate of Pull Requests, this PR increased your FTAR from 72.73 % to 75.00 %, good job, and for more quality improvements you can read about yellow findings.
	<br />
    -
    <details>
        <summary>More Details</summary>
        <br />
        <table>
            <tr>
                <td>Current FTAR</td>
                <td><b>75.00 %</b></td>
            </tr>
            <tr>
                <td>Total Reviewed Pull Requests for the last 4 weeks</td>
                <td>12</td>
            </tr>
            <tr>
                <td>Passed</td>
                <td>9</td>
            </tr>
            <tr>
                <td>Rejected</td>
                <td>3</td>
            </tr>
        </table>
        More details about how FTAR is calculated, refer to this page - <a
            href="https://confluence.devfactory.com/x/Cw8dGg">FTAR - First time acceptance rate</a>
    </details>

</details>

**Comment by alexander-a-chernyshev** [comment]
Hi, I'm Alexander Chernyshev from [Aurea Central Reviewers Team](https://github.com/orgs/trilogy-group/teams/aurea-central-reviewers/members) and I have just reviewed your PR.
<b>How likely are you to recommend the Code Review service to another developer?</b>

Not at all likely \[ [1](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/e2e-pltfrm-examples/pull/1&entry.1163746017=1) \] \[ [2](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/e2e-pltfrm-examples/pull/1&entry.1163746017=2) \] \[ [3](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/e2e-pltfrm-examples/pull/1&entry.1163746017=3) \] \[ [4](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/e2e-pltfrm-examples/pull/1&entry.1163746017=4) \] \[ [5](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/e2e-pltfrm-examples/pull/1&entry.1163746017=5) \] \[ [6](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/e2e-pltfrm-examples/pull/1&entry.1163746017=6) \] \[ [7](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/e2e-pltfrm-examples/pull/1&entry.1163746017=7) \] \[ [8](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/e2e-pltfrm-examples/pull/1&entry.1163746017=8) \] \[ [9](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/e2e-pltfrm-examples/pull/1&entry.1163746017=9) \] \[ [10](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/e2e-pltfrm-examples/pull/1&entry.1163746017=10) \] Very likely

> TIP: Press `Ctrl` (or `⌘` in a Mac) to open the survey in a new tab.

**Pull request merged**